tests/test_base_device.py

```diff
@@ -633,6 +633,65 @@ def test_num_wires_property(self):
         assert dev.num_wires == len(wires)
 
 
+class TestTrackerFunctionality:
+    def test_tracker_batched(self):
+        """Test that the tracker works for batched circuits"""
+        dev = qml.device("default.qubit", wires=1, shots=100)
+        qiskit_dev = QiskitDevice2(wires=1, backend=AerSimulator(), shots=100)
+
+        x = np.array(0.1, requires_grad=True)
+
+        @qml.qnode(dev, diff_method="parameter-shift")
+        def circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.Z(0))
+
+        @qml.qnode(qiskit_dev, diff_method="parameter-shift")
+        def qiskit_circuit(x):
+            qml.RX(x, wires=0)
+            return qml.expval(qml.Z(0))
+
+        with qml.Tracker(dev) as tracker:
+            qml.grad(circuit)(x)
+
+        with qml.Tracker(qiskit_dev) as qiskit_tracker:
+            qml.grad(qiskit_circuit)(x)
+
+        assert qiskit_tracker.history["batches"] == tracker.history["batches"]
+        assert tracker.latest.keys() == qiskit_tracker.latest.keys()
+        assert tracker.history.keys() == qiskit_tracker.history.keys()
+        assert tracker.history["shots"] == qiskit_tracker.history["shots"]
+
+        print(qiskit_tracker.history["batches"])
+        print(tracker.history["batches"])
+
+        # TODO: qiskit_tracker.history["results"] has a format that is inconsistent
+        # to tracker.history["results"], but this is due to the post processing
```

Comment:
Is there an easy way around it or to manually revert the `post-processing` changes? 🤔

Reply:
Or maybe look at the `simulator_tracking` and see if we could modify it for the qiskit device?

Reply:
> Is there an easy way around it or to manually revert the `post-processing` changes? 🤔

Maybe there is an easy way around it but I couldn't find one 😅 

> Or maybe look at the simulator_tracking and see if we could modify it for the qiskit device?

I'll look into this...